key/stringify.go

```diff
@@ -46,19 +47,37 @@ func Stringify(key interface{}) (string, error) {
 		str = key
 	case map[string]interface{}:
 		keys := SortedKeys(key)
-		for _, k := range keys {
+		var err error
+		for i, k := range keys {
 			v := key[k]
-			if len(str) > 0 {
-				str += "_"
-			}
-			s, err := Stringify(v)
+			keys[i], err = Stringify(v)
```

Comment:
Because we assume that `map[string]interface{}` is only used for composite values, which have a fixed set of attribute names, so we don't need to put the names in there.  For a collection (i.e., `map[Key]interface{}`) we need the keys in there otherwise it's hard to make sense of what is what.

We could add the attribute names to the string representation for the composite value case, I don't think it would hurt much, except that it would just change all the existing paths that contain the string representation of a composite value.